src/scimlfunctions.jl

```diff
@@ -1126,33 +1126,39 @@ RODEFunction{iip,recompile}(f;
                            syms = __has_syms(f) ? f.syms : nothing,
                            indepsym= __has_indepsym(f) ? f.indepsym : nothing,
                            colorvec = __has_colorvec(f) ? f.colorvec : nothing,
-                           sys = __has_sys(f) ? f.sys : nothing)
+                           sys = __has_sys(f) ? f.sys : nothing,
+                           analytic_full = __has_analytic_full(f) ? f.analytic_full : false)
 ```
 
 Note that only the function `f` itself is required. This function should
-be given as `f!(du,u,p,t)` or `du = f(u,p,t)`. See the section on `iip`
+be given as `f!(du,u,p,t,W)` or `du = f(u,p,t,W)`. See the section on `iip`
 for more details on in-place vs out-of-place handling.
 
 All of the remaining functions are optional for improving or accelerating
 the usage of `f`. These include:
 
-- `mass_matrix`: the mass matrix `M` represented in the ODE function. Can be used
-  to determine that the equation is actually a differential-algebraic equation (DAE)
-  if `M` is singular. Note that in this case special solvers are required, see the
-  DAE solver page for more details: https://diffeq.sciml.ai/stable/solvers/dae_solve/.
-  Must be an AbstractArray or an AbstractSciMLOperator.
-- `analytic(u0,p,t)`: used to pass an analytical solution function for the analytical
-  solution of the ODE. Generally only used for testing and development of the solvers.
-- `tgrad(dT,u,p,t)` or dT=tgrad(u,p,t): returns ``\frac{\partial f(u,p,t)}{\partial t}``
-- `jac(J,u,p,t)` or `J=jac(u,p,t)`: returns ``\frac{df}{du}``
-- `jvp(Jv,v,u,p,t)` or `Jv=jvp(v,u,p,t)`: returns the directional derivative``\frac{df}{du} v``
-- `vjp(Jv,v,u,p,t)` or `Jv=vjp(v,u,p,t)`: returns the adjoint derivative``\frac{df}{du}^\ast v``
+- `mass_matrix`: the mass matrix `M` represented in the RODE function. Can be used
+  to determine that the equation is actually a random differential-algebraic equation (RDAE)
+  if `M` is singular. Currently, however, all solvers assume `M` is the identity and do not take this term into account. Special solvers still need to be written to use a non-identity mass matrix.
```

Comment:
```suggestion
  if `M` is singular.
```

"Documentation at a distance" is hard to maintain, so we generally don't do that. The solvers should just error if it's not supported (I would think they already do? Worth checking)